.github/workflows/rust.yml

```diff
@@ -53,6 +73,16 @@ jobs:
         # the binary will be used by the next cargo test step
         run: cargo build
       - name: Run cargo tests
-        run: cargo test --all-targets --all-features
+        env:
+          GITHUB_ACTIONS_BASH_CONTAINER: 1
+        run: |
+          docker run --volume $PWD:/root --network host --detach --name test_container ${{ matrix.bash_docker_image }} sleep infinity
+          # coreutils includes mktemp
+          docker exec test_container apk add coreutils curl
+          if [ "${{ matrix.sed }}" == "GNU sed" ]; then
+            docker exec test_container apk add sed
+          fi
```

Comment:
a bunch of things are wrong with this

1. dependencies are added into a docker container in the build stage (dockerfile)
2. there is no reason to expose the container to the host network
3. i suppose it will write as root to the PWD, which is usually not a very good idea since it will cause dubious ownership. the correct way is to run it as the current uid&gid inside docker
3.1. (maybe) use docker volumes instead. its a usually cleaner way to mount something in a container, but it depends on a specific case and im not sure what exactly is happening there
4. you have set `sleep inf` as entrypoint, which is a wrong understanding of how docker works. i suppose you want to `cargo test --all-targets --all-features` inside docker, so you should set that as the entrypoint

Reply:
1. If other maintainers agree, I would create a Dockerfile after they're choosing a registry to store the image (Docker Hub or GitHub Cotainer registory(ghcr)). But maintaining an image requires another sort of maintaining resource and this building contains just two command, I used the inline commands. If your concern is the speed of testing, I will try caching the image with `docker write` and `actions/cache`.
2. That was because of https://github.com/amber-lang/amber/blob/82817af9de66eec4b6e32aa3a0f65007fbcdf854/src/tests/extra.rs#L65-L68 test which starts a web server and tries to download a file from it. I didn't want to i) install rust in a bash image or ii) install bash in the rust image. If cleaner solution is needed, we could set up a web server instead of rust code, maybe using docker, and share their network, then do the test.
3. I agree. I saw that junk files are created in $PWD on my laptop. I will fix, thank you.
4. (I will respond as a reply of the below comment)

Reply:
I am against a Docker image for our needs to maintain etc

Reply:
> I saw that junk files are created in $PWD on my laptop.

I've realized I didn't need to mount $PWD at all, so I've removed it.

Reply:
> 1. I would create a Dockerfile after they're choosing a registry to store the image

there's no need for that, you can have the dockerfile in the repo and just build from it when you have to.

```bash
$ cat Dockerfile
FROM bash:4.5
RUN apk add something
$ docker build -t testbash . # will use $PWD/Dockerfile
$ docker run --name tests --remove testbash cargo test
```

Reply:
I don't think that there is a majority that want a dockerfile for the Amber project right now, so I would ignore that suggestion right now.